builtin/logical/transit/path_export.go

```diff
@@ -187,22 +186,32 @@ func getExportKey(policy *keysutil.Policy, key *keysutil.KeyEntry, exportType st
 			return strings.TrimSpace(base64.StdEncoding.EncodeToString(key.Key)), nil
 
 		case keysutil.KeyType_RSA2048, keysutil.KeyType_RSA3072, keysutil.KeyType_RSA4096:
-			return encodeRSAPrivateKey(key.RSAKey), nil
+			return encodeRSAPrivateKey(key), nil
 		}
 	}
 
 	return "", fmt.Errorf("unknown key type %v", policy.Type)
 }
 
-func encodeRSAPrivateKey(key *rsa.PrivateKey) string {
+func encodeRSAPrivateKey(key *keysutil.KeyEntry) string {
 	// When encoding PKCS1, the PEM header should be `RSA PRIVATE KEY`. When Go
 	// has PKCS8 encoding support, we may want to change this.
-	derBytes := x509.MarshalPKCS1PrivateKey(key)
-	pemBlock := &pem.Block{
-		Type:  "RSA PRIVATE KEY",
+	var blockType string
+	var derBytes []byte
+	if !key.IsPrivateKeyMissing() {
+		blockType = "RSA PRIVATE KEY"
+		derBytes = x509.MarshalPKCS1PrivateKey(key.RSAKey)
+	} else {
+		blockType = "RSA PUBLIC KEY"
+		derBytes = x509.MarshalPKCS1PublicKey(key.RSAPublicKey)
```

Comment:
Not quite sure I agree with this change @stevendpclark btw. I think we should probably continue exporting only the private key and not automatically switch to the public key if the private key is not available, and instead err out.

Reply:
Now that I think about it I am also not sure. I see at least three sensible ways of what the `/export` endpoint can return when a key version doesn't have a private key:
- Return an error
- Return the public key (currently implemented in the PR)
- Return an empty string

As both return public key and empty string change the behavior of what is expected to be returned depending on the key to export, I think they are prone to confusion and errors unless they are well documented.
With that, returning an error seems the most valid approach.

Reply:
@Gabrielopesantos Yeah, @stevendpclark and I went looking a bit on Friday to try and understand this question better and we got to that it was just a quirk of how things were done. I forget exactly what we decided, but I think we thought leaving it private-key only was fine.

Steve, do you recall if we discussed adding a new `public` type export and maybe a `both` type export? Not sure what you thought of that? 

Regardless, I think we're fine merging this as-is, everything else here are just minor nits :-)